tests/test_setup.py

```diff
@@ -0,0 +1,31 @@
+"""test_setup.py.
+
+Unit tests for testing initial setup. The intention is these tests won't be
+part of the main test suite, and will only by run as needed.
+
+TODO: make this run 'on request' only.
+"""
+
+import pytest
+
+
+class TestSetup:
+    """A class of tests to check correct set-up.
+
+    A set of tests to:
+        1. check `heimdall_transport` is installed.
+    """
+
+    @pytest.mark.setup
+    def test_heimdall_transport_install(self) -> None:
+        """Check `heimdall_transport is installed.
+
+        This is a simple test to check the package has been installed and is
+        available for use.
+        """
+        try:
+            import heimdall_transport as ht
+
+            assert ht.__name__ == "heimdall_transport"
+        except ImportError:
+            pytest.fail("Unable to find `heimdall_transport`.")
```

Comment:
Failing for me:

```
>>> import heimdall_transport as ht
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'heimdall_transport'
>>> import src.heimdall_transport as ht
>>> ht.__name__
'src.heimdall_transport'
```
I didn't expect to see a src folder any more. Suggest ditching it in favour of `heimdall_transport`.

Reply:
The src -> package folder structure is a standard pypi layout: https://packaging.python.org/en/latest/tutorials/packaging-projects/

Have you called `pip install -e .` before running pytest? I would expect pytest to fail before this is done, and then work afterwards. It's a check to make sure `heimdall_transport` has been installed and is available.